src/Lumi/Components/Button.purs

```diff
@@ -130,11 +138,19 @@ iconButton = makeStateless iconComponent render
         , "data-color": props.color
         , "data-size": show props.size
         , "data-testid": props.testId
-        , disabled: props.disabled
         , onClick: props.onPress
         , style: props.style
         , type: props.type
-        , "data-loading": props.loading
+        , disabled:
+            case props.buttonState of
+              Enabled -> false
+              Disabled -> true
+              Loading -> true
+        , "data-loading":
+            case props.buttonState of
+              Enabled -> false
+              Disabled -> true
```

Comment:
Yes, the API looks good to me - although (sorry) I only just noticed that this will set `data-loading` to `true` when the button state is `Disabled`, which seems slightly odd. Should `disabled` and `data-loading` always have the same value?

Reply:
I believe so, as we want our loading button to reflect the disabled style guidelines:

<img width="123" alt="Screen Shot 2019-06-26 at 3 32 17 PM" src="https://user-images.githubusercontent.com/632981/60184112-c2271b00-9827-11e9-8502-0b505908cbc8.png">

However, in the CSS we do have this duplicate then:

`"&:disabled, &[data-loading=\"true\"]":
              { backgroundColor: cssStringHSLA colors.primary2
              , cursor: "default"
              }`

Reply:
Ah, makes sense, thanks! I think that's probably not worth worrying too much about just now; I've probably held this PR up too long already anyway.